Add ignore_docm option

dufeiyu opened the pull request:
For non-cle analysis, docm variants are not needed and this PR is to add ignore_docm option to docm_filter.pl to filter out all docm variants.

dufeiyu committed:
Add ignore_docm option

chrisamiller commented on docm_filter.pl:
```diff
@@ -32,6 +32,9 @@
     }
     elsif (/^#CHROM/) {
+        if ($ignore_docm_flag and $ignore_docm_flag eq 'ignoreDOCM') {
+            say $docm_filter_fh '##FILTER=<ID=IgnoreDOCM,Description="ignore Docm variants">';
```
From the perspective of someone looking at the FILTER field in the final VCF, I like DOCM_ONLY as the name of the filter (which answers the question "why was this filtered?")

dufeiyu replied:
yeah, it makes sense. Change is made.

chrisamiller reviewed (approved):
other than the renaming request, looks good!

dufeiyu committed:
Change filter name to DOCM_ONLY

chrisamiller committed:
change outdir to outfile

I'm working on the CWL, and it makes sense to let the user specify the full output filename, not just the directory.

chrisamiller commented:
Requesting a re-review, since I pushed a minor change here

chrisamiller committed:
also change the flag to a 0/1 flag, instead of a string

chrisamiller committed:
fixing the ignore flag

tmooney commented on docm_filter.pl:
```diff
@@ -5,8 +5,8 @@
-die("Wrong number of arguments. Provide docm_out_vcf, normal_cram, tumor_cram, output_dir") unless scalar(@ARGV) == 4;
-my ($docm_out_vcf, $normal_cram, $tumor_cram, $outdir) = @ARGV;
+die("Wrong number of arguments. Provide docm_out_vcf, normal_cram, tumor_cram, output_vcf_file") unless scalar(@ARGV) == 4;
+my ($docm_out_vcf, $normal_cram, $tumor_cram, $outfile, $ignore_docm_flag) = @ARGV;
```
It's not possible for the `$ignore_docm_flag` to be set, since the previous line mandates only four parameters be specified.

chrisamiller committed:
fixing number of params, flag name

chrisamiller commented:
fixed that - thanks, Tom!

tmooney reviewed (approved):
+1, although merging this will immediately break existing workflows, since this line is not versioned:

https://github.com/genome/analysis-workflows/blob/5a4fed24f01f6d7c7ce1f595b272d82d97d4f9bd/definitions/tools/somatic_docm_filter.cwl#L9

chrisamiller commented:
Not planning to merge it until the complete replacement is ready (that merges docm in the new way, etc).  That PR is still pending getting a run of the pipeline through on real data, since the chr17 subset doesn't contain any docm sites.  That has been a little more problematic than I'd like.
